dom/data/core.js

```diff
@@ -0,0 +1,64 @@
+var isEmptyObject = require("../../js/is-empty-object/is-empty-object");
+
+var data = {};
+var expando = "can" + new Date();
+var uuid = 0;
+
+var setData = function(name, value) {
+	var id = this[expando] || (this[expando] = ++uuid),
+		store = data[id] || (data[id] = {});
+	if (name !== undefined) {
+		store[name] = value;
+	}
+	return store;
+};
+
+// delete this node's `data`
+// returns the number of keys that were deleted
+var deleteNode = function() {
+	var id = this[expando];
+	var count = 0;
+	if(id) {
+		count = data[id] ? Object.keys(data[id]).length : 0;
+		delete data[id];
+	}
+	return count;
+};
+
+/*
+ * Core of domData that does not depend on mutationDocument
+ * This is separated in order to prevent circular dependencies
+ */
+module.exports = {
+	_data: data,
+
+	getCid: function() {
+		return this[expando];
+	},
+
+	cid: function(){
+		return this[expando] || (this[expando] = ++uuid);
+	},
+
+	expando: expando,
+
+	get: function(key) {
+		var id = this[expando],
+			store = id && data[id];
+		return key === undefined ? store || setData(this) : store && store[key];
+	},
+
+	set: setData,
+
+	clean: function(prop) {
+		var id = this[expando];
+		if (data[id] && data[id][prop]) {
```

Comment:
I can update that, but this is how it is in the existing code: https://github.com/canjs/can-util/blob/master/dom/data/data.js#L72-L74. I just moved it over into `core`.